plugins/EventGhost/__init__.py

```diff
@@ -595,55 +595,292 @@ def __call__(self):
+has finished processing.
+<br><br>
+* ***Restore eg.event***
+  ______
+  If add to queue has not been checked this option will appear. This
+relates more to the scripting portions of EventGhost. What this done is
+each and every time an event gets triggered there are 2 variables that
+get set into place. Those 2 variables are eg.event and eg.eventString.
+When you trigger an event while the current event is running those 2
+variables will get changed to the new event. Upon completion of the new
+event if you would like to change those variables back to the event that
+ran this action then check this box.
+<br><br>
+* ***Using a Python Expression***
+  ______
+  You can use a python expression in several ways. The expression 
+**MUST** be wrapped in curly braces {}. This is the identifier that
+tells EventGhost that it needs to do some work.
+
+  You can pass global variables which are stored in eg.globals by
+wrapping the variable name in the curly braces.
+
+        {eg.globals.some_variable}
+
+  If you want to transfer the results of another action you can do this
+as well.
+
+        {eg.plugins.SomePlugin.SomeAction()}
+  
+  ***Or maybe you want to do something a little more complex.***
+  
+  A different value passed based on if a global is True or False.
+
+        {"TV.On" if eg.globals.tv_power else "TV.Off"}
+
+  Or checking a global for a specific value and passing True or False.
+
+        {eg.plugins.SomePlugin.SomeAction() == 100}
+
+  When using a python expression in a payload the curly braces are the
+same thing that is used in a dictionary but our crafty programmers have
+accounted for this so don't worry.
+
+  These expressions get run when the TriggerAction gets run. So if you
+have a programmed wait time (see below) the data may be different at the
+start of the wait time then at the end.
+  
+"""
 class TriggerEvent(eg.ActionBase):
+    __doc__ = DESCRIPTION
     name = "Trigger Event"
-    description = (
-        "Triggers an event with an optional delay."
-    )
+    description = DESCRIPTION
     iconFile = "icons/Plugin"
 
     class text:
-        labelWithTime = 'Trigger event "%s" after %.2f seconds'
-        labelWithoutTime = 'Trigger event "%s"'
         text1 = "Event string to fire:"
         text2 = "Delay the firing of the event:"
         text3 = "seconds. (0 = fire immediately)"
+        text4 = "Add event to event queue:"
+        text5 = "Return eg.event to original event:"
+        text6 = "Event Payload:"
+
+    def __call__(
+        self,
+        eventString,
+        waitTime=0,
+        payload=None,
+        queueEvent=True,
+        restoreEvent=False
+    ):
+
+        def parse(value):
+            if value is None:
+                return None
+            parsed_value = eg.ParseString(value)
+            if value == parsed_value:
+                try:
+                    value = eval(value)
+                except SyntaxError:
```

Comment:
Here i get a NameError. With `except (SyntaxError, NameError):` it looks good to me.